cvat/apps/dataset_manager/formats/datumaro.py

```diff
@@ -15,29 +14,23 @@
 
 from .registry import dm_env, exporter, importer
 
-class DeleteImagePath(ItemTransform):
-    def transform_item(self, item):
-        image = None
-        if item.has_image and item.image.has_data:
-            image = Image(data=item.image.data, size=item.image.size)
-        return item.wrap(image=image, point_cloud='', related_images=[])
-
 
 @exporter(name="Datumaro", ext="ZIP", version="1.0")
 def _export(dst_file, temp_dir, instance_data, save_images=False):
     dataset = Dataset.from_extractors(GetCVATDataExtractor(
         instance_data=instance_data, include_images=save_images), env=dm_env)
-    if not save_images:
-        dataset.transform(DeleteImagePath)
     dataset.export(temp_dir, 'datumaro', save_images=save_images)
 
     make_zip_archive(temp_dir, dst_file)
 
-@importer(name="Datumaro", ext="ZIP", version="1.0")
+@importer(name="Datumaro", ext="JSON, ZIP", version="1.0")
 def _import(src_file, temp_dir, instance_data, load_data_callback=None, **kwargs):
-    Archive(src_file.name).extractall(temp_dir)
-
-    dataset = Dataset.import_from(temp_dir, 'datumaro', env=dm_env)
+    if zipfile.is_zipfile(src_file):
+        zipfile.ZipFile(src_file).extractall(temp_dir)
+        dataset = Dataset.import_from(temp_dir, 'datumaro', env=dm_env)
+    else:
+        load_data_callback = None
```

Comment:
As I can see `load_data_callback` is used only for importing project datasets so `zipfile.is_zipfile(src_file)` will always be true and it is not clear to me why we need to set the `load_data_callback` explicitly to `None`. Am I missing something?

Reply:
It's true, but with this implementation there is no need to think about how the function is called.

Reply:
It's true, but with this implementation there is no need to think about how the function is called.

Reply:
Maybe in this case we need to raise an exception if `load_data_callback` was passed with json file. (since it is unexpected behavior)